modules/apps/jaggery/mdm/src/mdm/modules/user.js

```diff
@@ -62,26 +63,38 @@ var userModule = function () {
      *  'content': {}
      * }
      */
-    privateMethods.callBackend = function(url, method) {
-        var url = mdmProps["httpsURL"] + url;
-        var xhr = new XMLHttpRequest();
-        xhr.open(method, url);
-        xhr.send();
-
-        var response = {};
-        if (xhr.status == 200 && xhr.readyState == 4) {
-            var responsePayload = parse(xhr.responseText);
-            response.content = responsePayload["responseContent"];
-            response.status = "success";
-            return response;
-        } else {
-            response.content = xhr.responseText;
-            response["status"] = "error";
-            return response;
+    privateMethods.callBackend = function (url, method) {
+        switch (method) {
+            case constants.HTTP_GET:
+                var response = serviceInvokers.XMLHttp.get(url, function (responsePayload) {
+                                                               var response = {};
+                                                               response.content = responsePayload["responseContent"];
+                                                               response.status = "success";
+                                                               return response;
+                                                           },
+                                                           function (responsePayload) {
+                                                               var response = {};
+                                                               response.content = responsePayload;
+                                                               response.status = "error";
+                                                               return response;
+                                                           });
+                return response;
+                break;
+            case constants.HTTP_POST:
+                //todo
```

Comment:
This Method only supports GET request currently.this was added to extend the method when needed.As per the discussion had offline recommended changes are applied.
Recommended fixes can be found in PR[1]

[1]. https://github.com/wso2/product-mdm/pull/166